base/src/main/java/org/kontinuity/catapult/base/identity/UserPasswordIdentity.java

```diff
@@ -0,0 +1,29 @@
+package org.kontinuity.catapult.base.identity;
+
+/**
+ * @author <a href="mailto:ggastald@redhat.com">George Gastaldi</a>
+ */
+public class UserPasswordIdentity implements Identity {
+
+    private final String username;
+    private final String password;
+
+    UserPasswordIdentity(String username, String password) {
+        this.username = username;
```

Comment:
IdentityFactory is the only place to create UserPasswordIdentity objects. Maybe move the precondition checking here?